crates/apps/src/lib/node/ledger/shell/process_proposal.rs

```diff
@@ -1815,7 +1456,7 @@ mod test_process_proposal {
             Err(TestError::RejectProposal(response)) => {
                 assert_eq!(
                     response[0].result.code,
-                    u32::from(ResultCode::AllocationError)
+                    u32::from(ResultCode::TxGasLimit)
```

Comment:
We made the tx use more than the amount of gas allowed in a block so that this could happen. In theory, we could have done multiple txs whose gasses summed up to more than the block amount, but it doesn't seem to add much value.